crates/task-impls/src/consensus.rs

```diff
@@ -1063,11 +1067,20 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES>, A: ConsensusApi<TYPES, I> +
                     .await;
 
                 // Add to the storage that we have received the VID disperse for a specific view
+                if let Err(e) = self.storage.write().await.append_vid(disperse).await {
+                    error!(
+                        "Failed to store VID Disperse Proposal with error {:?}, aborting vote",
+                        e
+                    );
+                    return;
+                }
+
                 self.consensus
                     .write()
                     .await
                     .vid_shares
                     .insert(view, disperse.clone());
```

Comment:
The gist of this work is to add a trait for persistent storage with a result to allow us to detect things like disk capacity being exceeded, among other things. The initial phase of work is to only store DA and VID proposals, as we see here. But I noticed that the `Consensus` type actually *also* stores these things, but in memory. The question I'm getting at is: should we store both? Right now the type only supports `append`, so my thought it yes, but I wanted to highlight this to make sure I'm not missing something here. cc @jbearer @bfish713

Reply:
I think we want both, consensus needs wants a quick in memory store for it's state which we can garbage collect and modify without worrying about what effect it might have on the application.

Reply:
The inverse is also true, the application might want to mess with the data in ways that would break what consensus wants for voting and it should be free to do so.  That's my 2 cents at least

Reply:
Totally agree.

The one thing I think we _can_ get rid of is the `PayloadStore` (is it still called that?) where we store the full payloads as a DA replica. I believe this is only used in consensus to populate the `Option<Payload>` field of the leaf in decide events. We could potentially remove this field and the logic to populate it on Decide, since the application is no capable of retrieving the payload itself

Reply:
Tracking: https://github.com/EspressoSystems/HotShot/issues/2808